rpc/src/client/transport/http.rs

```diff
@@ -73,33 +73,30 @@ impl Builder {
     /// attempt to connect using the [HTTP CONNECT] method.
     ///
     /// [HTTP CONNECT]: https://en.wikipedia.org/wiki/HTTP_tunnel
-    pub fn proxy_url(&mut self, url: HttpClientUrl) -> &mut Self {
+    pub fn proxy_url(mut self, url: HttpClientUrl) -> Self {
         self.proxy_url = Some(url);
         self
     }
 
     /// Try to create a client with the options specified for this builder.
-    pub fn build(&self) -> Result<HttpClient, Error> {
-        match &self.proxy_url {
+    pub fn build(self) -> Result<HttpClient, Error> {
+        match self.proxy_url {
             None => Ok(HttpClient {
                 inner: if self.url.0.is_secure() {
-                    sealed::HttpClient::new_https(self.url.clone().try_into()?)
+                    sealed::HttpClient::new_https(self.url.try_into()?)
                 } else {
-                    sealed::HttpClient::new_http(self.url.clone().try_into()?)
+                    sealed::HttpClient::new_http(self.url.try_into()?)
```

Comment:
Hm, OK. I did not think it worthwhile to change to the less usable self-consuming builder, just to save a few clone operations on small data structures.
Or is there a bigger reason?

Reply:
The `&mut pattern` does work very well for our use in the event monitor, and since that's the main consumer of the library I figured we could switch to a self-consuming one. It's slightly less flexible in the sense that one may have to re-assign it to variable when setting things conditionally but otherwise it makes for a more fluent API in our current use case. The lack of cloning is just a side benefit from consuming the builder at the end, as most builders do.